firedrake/slate/slate.py

```diff
@@ -821,6 +825,24 @@ def arguments(self):
         return self._args
 
 
+def space_equivalence(A, B):
+    """Checks that two function spaces are equivalent.
+
+    :arg A: A function space.
+    :arg B: Another function space.
+
+    Returns `True` if they have matching meshes, elements, and rank. Otherwise,
+    `False` is returned.
+    """
+
+    equiv = False
+    assert A.mesh() == B.mesh(), "Mismatching meshes!"
+    if A.rank == B.rank and A.ufl_element() == B.ufl_element():
```

Comment:
Under what circumstances do the elements match, but the rank not?

Reply:
This is just me being paranoid and making *extra* sure. But now that I think about it, rank is probably a superfluous check.